src/frequenz/quantities/_energy.py

```diff
@@ -174,15 +175,18 @@ def __truediv__(
         from ._power import Power  # pylint: disable=import-outside-toplevel
 
         match other:
-            case float():
-                return super().__truediv__(other)
+            case float() | Decimal():
+                return super().__truediv__(other)  # type: ignore[operator]
             case Energy():
                 return self._base_value / other._base_value
             case timedelta():
-                return Power._new(self._base_value / (other.total_seconds() / 3600.0))
+                return Power._new(
+                    self._base_value
+                    / self._base_value.__class__(other.total_seconds() / 3600.0)
+                )
```

Comment:
Shouldn't you use Decimals before division?
Instead of `Decimal(other.total_seconds() / 3600.0)` ->  `Decimal(other.total_seconds()) / Decimal(3600)`